content/administration/upgrade/odoo_sh.rst

```diff
@@ -1,66 +1,125 @@
-
 =======
 Odoo.sh
 =======
 
-Test upgrade request
-====================
+.. _odoo_sh_upgrade/overview:
 
-Download a dump of your database (from the :ref:`Builds view
-<odoosh-gettingstarted-builds-download-dump>`), choose the **exact copy** and **without filestore**
-options. Upload the .sql.gz dump on https://upgrade.odoo.com/upload and select the testing purpose.
-Once it's processed, you'll get a dump of the database in return.
+Overview
+========
 
-.. important::
-   To have the most up-to-date test version of your database, create a backup of your production
-   database just before creating your request. Upload it in your staging branch, select **Exact
-   dump** and **Without filestore**, and then click *start*.
+Odoo.sh is integrated with the upgrade platform to make the upgrade process easier.
+
+.. note::
+   The :guilabel:`Upgrade` tab is available in the branches view. It is only available for valid
+   projects with a valid production build.
+
+.. image:: odoo_sh/odoo-sh-menu.png
+   :align: center
+   :alt: Click on the upgrade menu
+
+The suggested upgrade steps on Odoo.sh are:
+
+#. On a :guilabel:`Development` branch, upgrade your custom modules to keep them compatible with the
+   new version and thoroughly **test them**.
+#. Switch that branch to the :guilabel:`Staging` branch, **upgrade** the last daily production
+   backup and **test it**. Write upgrade scripts if necessary.
+#. Trigger the production upgrade from your :guilabel:`Production` branch and sit tight.
 
-Test your upgraded database
+.. _odoo_sh_upgrade/custom-modules:
+
+Upgrade your custom modules
 ===========================
 
-At the beginning of your upgrade project, make sure that you create a new staging branch for testing
-purposes by forking your main branch.
+The first step is to upgrade your custom modules to keep them compatible with the new version. Fork
+your :guilabel:`Production` branch in the :guilabel:`Development` stage, then go to the settings of
+your :guilabel:`Development` branch and select the Odoo version you target. If needed, modify your
+code to be compatible with the new version. Make sure to **test** your features are still working
+correctly.
```

Comment:
I don't think the use of guilabel is 100% accurate. We use it when we refer to a specific element of the user interface. If you refer to a concept, a name, but not a graphical element, then it shouldn't be a guilabel.